Monads.NET/Maybe.cs

```diff
@@ -162,6 +197,37 @@ public static TInput IfDo<TInput>(this TInput o, Func<TInput, bool> evaluator, A
         }
 
         /// <summary>
+        /// Allows for a null safe action on a object.
+        /// </summary>
+        /// <typeparam name="TInput">Instance of object to be acted on</typeparam>
+        /// <param name="o">object to be extended</param>
+        /// <param name="action">function that will act on the object</param>
+        /// <returns>object that was acted on</returns>
+        public static Task DoAsync<TInput>(this TInput o, Func<TInput, Task> action)
+            where TInput : class
+        {
+            return (o == null) ? TaskEx.FromResult<object>(null) : action(o);
```

Comment:
Why not make this an `async` implementation and just say 
`if(o != null) await action(o);`